Qr feature

luisgg98 opened the pull request:
Now creating a qr code is available on the website and it has been tested using Mocks

Commit:
QR feature

I can test it by the command curl on terminal however if I try to use Mock
in Spring doesn't work.

CHECK: Revise automation

Commit:
can't load qr image

Once the response is received it can't load the image correctly.
I thinks it's because the byte array it's not correcty coded on 64base
I'm trying tomorrow

Commit:
QR code appears on screen

Commit:
cleaning

Commit:
Test working

It's that I added @Mock instead of @InjectMock to the QRController, that's why it wasn't wokring

Commit:
Checking url not null

luisgg98 committed:
Merge branch 'master' into qr-feature

Commit:
fix the error

Commit:
Merge branch 'qr-feature' of https://github.com/URLY-WebEngineering/UrlShortener into qr-feature

Commit:
changes

Commit:
Env variables set

Commit:
changing url controller

Commit:
More secure QRController

Commit:
I forgot to add one Mock

Commit:
TODOs done

jgarciapueyo committed:
Update .gitignore

Commit:
Remove dependencie

Commit:
delete data.png

jgarciapueyo commented:
I have revised the Pull Request. However, because I do not have experience with tests (unit test of controllers and services, or integration tests), I have asked @AlbertoCalvoRubio to finish revising the tests of your PR.

AlbertoCalvoRubio commented on src/main/java/urlshortener/web/QRController.java:
```diff
@@ -0,0 +1,50 @@
+
+  public QRController(SafeBrowsingService safeBrowsingService) {
+
+    this.safeBrowsingService = safeBrowsingService;
```
I think you don't use this service here

AlbertoCalvoRubio reviewed (changes requested):
- I have written a comment in the code. Please change that
- You should write some test about Service Testing at .. test/java/urlshortener/service , it may take a few lines
- We should wait for Pelicer's feedback about the bug pointed out by SonarCloud

Commit:
Delete not used service

Commit:
QR Service test added
I've added a test to the QR Service. This Junit verify if the substring
"data:image/png;base64" is in the string which contents the qrcode coded
in base64

luisgg98 commented:
I've corrected what I was doing wrong and removing the unnecessary code or files.
I think it's finally done. Thank you guys for your time and patience.

luisgg98 commented:
I've corrected what I was doing wrong and removed the unnecessary code or files.
I think it's finally done. Thank you guys for your time and patience.

AlbertoCalvoRubio commented:
Luis try to fix the vulnerability indicated by SonarCloud

luisgg98 commented:
According to what the teacher told us I have to make a lots of changes on the code, so I'll start a new PR soon
